hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/FileSourceBuilder.java

```diff
@@ -161,6 +168,31 @@ public FileSourceBuilder charset(@Nonnull Charset charset) {
                 SourceProcessors.readFilesP(directory, glob, sharedFileSystem, readFileFn));
     }
 
+    /**
+     * Builds a JSON file {@link BatchSource} with supplied components. The
+     * source expects a stream of JSON objects as the content of the file.
+     <p>
+     * The source does not save any state to snapshot. If the job is restarted,
+     * it will re-emit all entries.
+     * <p>
+     * Any {@code IOException} will cause the job to fail. The files must not
+     * change while being read; if they do, the behavior is unspecified.
+     * <p>
+     * The default local parallelism for this processor is 2 (or 1 if just 1
+     * CPU is available).
+     *
+     */
+    public <T> BatchSource<T> buildJson(@Nonnull Class<T> type) {
+        String charsetName = charset.name();
+        return build(path -> {
+            InputStreamReader reader = new InputStreamReader(new FileInputStream(path.toFile()), charsetName);
+            ValueIterator<T> valueIterator = JSON.std.beanSequenceFrom(type, reader);
```

Comment:
Maybe a JSON manipulation should be inside the `JsonUtil` class and Jackson jr not be exposed anywhere else?